client-core/src/wallet/default_wallet_client.rs

```diff
@@ -102,6 +103,21 @@ where
         passphrase: &SecUtf8,
         wallet_kind: WalletKind,
     ) -> Result<Option<Mnemonic>> {
+        // `estimate_password_strength` returns a score between `0-4`. Any score less than 3 should be considered too
+        // weak.
+        let password_entropy = estimate_password_strength(passphrase.unsecure(), &[name])
+            .chain(|| (ErrorKind::IllegalInput, "Blank passphrase"))?;
+
+        if password_entropy.score() < 3 {
+            return Err(Error::new(
+                ErrorKind::IllegalInput,
+                format!(
+                    "Weak passphrase: {}",
+                    parse_feedback(password_entropy.feedback().as_ref())
+                ),
```

Comment:
I guess this is probably fine, but not sure how annoying it'd be in quick testing -- we'll see, perhaps later there could be a "unsafe_development_only" flag to bypass that

Reply:
We can add following code in future to only build this in release builds:
```rust
if !cfg!(debug_assertions) {
    // Check password score only in non-debug builds
}
```